lighthouse-core/lib/dependency-graph/simulator/network-analyzer.js

```diff
@@ -185,6 +185,23 @@ class NetworkAnalyzer {
     });
   }
 
+  /**
+   * @param {LH.NetworkRequest[]} records
+   * @return {boolean}
+   */
+  static canTrustConnectionInformation(records) {
+    const connectionIdWasStarted = new Map();
+    for (const record of records) {
+      const started = connectionIdWasStarted.get(record.connectionId) || !record.connectionReused;
+      connectionIdWasStarted.set(record.connectionId, started);
+    }
+
+    // We probably can't trust the network information if all the connection IDs were the same
+    if (connectionIdWasStarted.size <= 1) return false;
```

Comment:
could this happen for a single request page? (or am I missing the nature of connections? :)

Reply:
most definitely, but in that case it won't matter since our inferred result will be identical to the real data